skyportal/tests/frontend/test_reminders.py

```diff
@@ -0,0 +1,192 @@
+import os
+import uuid
+import time
+
+from skyportal.tests import api
+from datetime import date, timedelta, datetime
+from selenium.webdriver.common.keys import Keys
+
+
+def post_and_verify_reminder(endpoint, token):
+    reminder_text = str(uuid.uuid4())
+    next_reminder = datetime.utcnow() + timedelta(seconds=5)
+    reminder_delay = 1
+    number_of_reminders = 2
+    request_data = {
+        'text': reminder_text,
+        'next_reminder': next_reminder.strftime("%Y-%m-%dT%H:%M:%S"),
+        'reminder_delay': reminder_delay,
+        'number_of_reminders': number_of_reminders,
+    }
+
+    status, data = api(
+        'POST',
+        endpoint,
+        data=request_data,
+        token=token,
+    )
+    assert status == 200
+    assert data['status'] == 'success'
+
+    status, data = api('GET', endpoint, token=token)
+    assert status == 200
+    assert data['status'] == 'success'
+    data = data['data']['reminders']
+    assert data[-1]['text'] == reminder_text
+    assert data[-1]['next_reminder'] == next_reminder.strftime("%Y-%m-%dT%H:%M:%S")
+    assert data[-1]['reminder_delay'] == reminder_delay
+    assert data[-1]['number_of_reminders'] == number_of_reminders
+
+    time.sleep(20)  # wait for the reminder to be sent
+    status, data = api('GET', endpoint, token=token)
+    assert status == 200
+    assert data['status'] == 'success'
+    data = data['data']['reminders']
+    assert len(data) == 1
+    assert data[-1]['text'] == reminder_text
+    assert data[-1]['next_reminder'] == (
+        next_reminder + timedelta(days=reminder_delay)
+    ).strftime("%Y-%m-%dT%H:%M:%S")
+    assert data[-1]['reminder_delay'] == reminder_delay
+    assert data[-1]['number_of_reminders'] == number_of_reminders - 1
+    return reminder_text
+
+
+def post_and_verify_reminder_frontend(driver, reminder_text):
+    search_button_xpath = driver.wait_for_xpath(
+        '//button[@data-testid="Search-iconButton"]'
+    )
+    driver.scroll_to_element_and_click(search_button_xpath)
+    search_bar = driver.wait_for_xpath('//input[@aria-label="Search"]')
+
+    search_bar.send_keys(f"{reminder_text}")
+    driver.wait_for_xpath(f'//*[text()="{reminder_text}"]', timeout=10)
+    search_bar.clear()
+
+    driver.scroll_to_element_and_click(
+        driver.wait_for_xpath('//*[@data-testid="AddIcon"]')
+    )
+    reminder_text_2 = str(uuid.uuid4())
+    driver.wait_for_xpath('//*[@id="root_text"]').send_keys(reminder_text_2)
+    next_reminder = (datetime.now() + timedelta(days=1)).strftime("%m/%d/%YT%I:%M %p")
+    driver.wait_for_xpath('//*[@id="root_next_reminder"]').send_keys(
+        next_reminder[0:11]
+    )
+    driver.wait_for_xpath('//*[@id="root_next_reminder"]').send_keys(Keys.TAB)
+    driver.wait_for_xpath('//*[@id="root_next_reminder"]').send_keys('01:01')
```

Comment:
Is this the right time?

Reply:
The time doesn't really matter for this part of the test. Here, we don't verify if the reminder gets triggered or not, just if it gets added and that we receive the notification telling us a new reminder was created.